core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -29,9 +30,9 @@
  *
  * @param <T> the type of resource this pool manages
  */
-// TODO(peis): Implements this from the Pool interface.
 @ThreadSafe
-public abstract class ResourcePool<T> {
+public abstract class ResourcePool<T> implements Pool<T> {
```

Comment:
@apc999 It was not implemented via Pool interface because Pool interface is added after this class. Pool interface was added for DynamicResourcePool